inference/fim-inference.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import gzip
+import json
+import importlib
+from pathlib import Path
+from tqdm import tqdm
+
+FIM_PREFIX = "<fim-prefix>"
+FIM_MIDDLE = "<fim-middle>"
+FIM_SUFFIX = "<fim-suffix>"
+FIM_PAD = "<fim-pad>"
+EOD = "<|endoftext|>"
+SPEC_TOKS = [EOD, FIM_PREFIX, FIM_MIDDLE, FIM_SUFFIX, FIM_PAD]
+
+def read_json_gz(p: Path):
```

Comment:
This function is also defined here, which you can import as `import ..util`

https://github.com/nuprl/MultiPL-E/blob/main/util.py